src/Orleans/Core/GrainFactory.cs

```diff
@@ -189,56 +219,73 @@ public Task<TGrainObserverInterface> CreateObjectReference<TGrainObserverInterfa
         /// </typeparam>
         /// <param name="obj">The reference being deleted.</param>
         /// <returns>A <see cref="Task"/> representing the work performed.</returns>
-        public Task DeleteObjectReference<TGrainObserverInterface>(
-            IGrainObserver obj) where TGrainObserverInterface : IGrainObserver
+        public Task DeleteObjectReference<TGrainObserverInterface>(IGrainObserver obj)
+            where TGrainObserverInterface : IGrainObserver
         {
-            var interfaceType = typeof(TGrainObserverInterface);
-            if (!interfaceType.IsInterface)
-            {
-                throw new ArgumentException(
-                    string.Format(
-                        "The provided type parameter must be an interface. '{0}' is not an interface.",
-                        interfaceType.FullName));
-            }
+            return GrainReference.DeleteObjectReference(obj);
+        }
 
-            if (!interfaceType.IsInstanceOfType(obj))
+        #region IGrainObserver Methods
+
+        private static Delegate MakeCreateObjectReferenceDelegate(Type interfaceType)
```

Comment:
`MakeCreateObjectReferenceDelegate` is also called only if we fail to find an invoker in `CreateObjectReference`.